docs/advanced.md

```diff
@@ -15,6 +15,42 @@ A Client instance is equivalent to a Session instance in `requests`.
 <Response [200 OK]>
 ```
 
+### Configuration
+
+Clients allow you to apply configuration to all outgoing requests by passing parameters to the `Client` constructor.
+
+For example, to apply a set of custom headers on every request:
+
+```python
+>>> url = 'http://httpbin.org/headers'
+>>> headers = {'user-agent': 'my-app/0.0.1'}
+>>> with httpx.Client(headers=headers) as client:
+...     r = client.get(url)
+...
+>>> r.json()['headers']['User-Agent']
+'app-1.0'
+```
+
+!!! note
+    When you provide a parameter at both the client and request levels, one of two things can happen:
+    
+    - For headers, query parameters and cookies, the values are merged into one.
+    - For all other parameters, the request-level value is used.
+
+Besides, `Client` constructor accepts some parameters that aren't available at the request level.
```

Comment:
I'm not a native english speaker but "besides" feels argumentative, I feel "additionally" may read better?

Maybe something like "Additionally, the `Client` constructor..."